webdav3/client.py

```diff
@@ -235,18 +235,22 @@ def valid(self):
         return True if self.webdav.valid() else False
 
     @wrap_connection_error
-    def list(self, remote_path=root):
+    def list(self, remote_path=root, getinfo = False):
         """Returns list of nested files and directories for remote WebDAV directory by path.
         More information you can find by link http://webdav.org/specs/rfc4918.html#METHOD_PROPFIND
 
         :param remote_path: path to remote directory.
-        :return: list of nested file or directory names.
+        :param getinfo: path and element info to remote directory, like cmd 'ls -l'.
+        :return: list of nested file or directory names and info.
         """
         directory_urn = Urn(remote_path, directory=True)
         if directory_urn.path() != Client.root and not self.check(directory_urn.path()):
             raise RemoteResourceNotFound(directory_urn.path())
 
         response = self.execute_request(action='list', path=directory_urn.quote())
+        if getinfo:
+            return WebDavXmlUtils.parse_get_list_info_response(response.content)
```

Comment:
@huangganggui here there is a defect due to the returning value contains the parent folder, you can see on the line 257 how it is filtered.